torch/utils/data/communication/protocol.py

```diff
@@ -62,80 +62,126 @@ def get_new_request(self, block=False):
             raise EmptyQueue('queue is empty')
         self._req_received = response
         return response
-
         # TODO: Validate supported requests
 
-    def response_reset(self):
+    def response_terminate(self):
         if not self.have_pending_request():
             raise Exception("Attempting to reply with pending request")
-        if not isinstance(self._req_received, communication.messages.ResetIteratorRequest):
+        if not isinstance(self._req_received, communication.messages.TerminateRequest):
             raise Exception(
-                "Replaying with reset status to other type of message")
-        self.response_queue.put(communication.messages.ResetIteratorResponse())
+                "Replaying with terminate status to other type of message")
+        self.response_queue.put(communication.messages.TerminateResponse())
         self._req_received = None
 
-    def response_next(self, value):
-        if not self.have_pending_request():
-            raise Exception("Attempting to reply with pending request")
-        self.response_queue.put(communication.messages.GetNextResponse(value))
-        self._req_received = None
 
-    def response_stop(self):
+class MapDataPipeQueueProtocolServer(ProtocolServer):
+    def response_item(self, key, value):
         if not self.have_pending_request():
             raise Exception("Attempting to reply with pending request")
-        self.response_queue.put(communication.messages.StopIterationResponse())
+        self.response_queue.put(communication.messages.GetItemResponse(key, value))
         self._req_received = None
 
-    def response_invalid(self):
+    def response_len(self, size):
         if not self.have_pending_request():
             raise Exception("Attempting to reply with pending request")
-        self.response_queue.put(communication.messages.InvalidStateResponse())
+        self.response_queue.put(communication.messages.LenResponse(size))
         self._req_received = None
 
-    def response_terminate(self):
+    def response_stop_iteration(self):
```

Comment:
This handles the case when index goes out of bound.

Reply:
Please name it differently